app/models/manageiq/providers/cloud_manager/auth_key_pair.rb

```diff
@@ -33,18 +33,13 @@ def self.create_key_pair_queue(userid, ext_management_system, options = {})
 
   def self.create_key_pair(ems_id, options)
     raise ArgumentError, _("ems cannot be nil") if ems_id.nil?
+
     ext_management_system = ExtManagementSystem.find(ems_id)
-    raise ArgumentError, _("ems cannot be found") if ext_management_system.nil?
 
     klass = ext_management_system.class_by_ems(:AuthKeyPair)
-    # TODO(maufart): add cloud_tenant to database table?
-    created_key_pair = klass.raw_create_key_pair(ext_management_system, options)
-    klass.create(
-      :name        => created_key_pair.name,
-      :fingerprint => created_key_pair.fingerprint,
-      :resource    => ext_management_system,
-      :auth_key    => created_key_pair.private_key
-    )
+
+    key_pair_opts = klass.raw_create_key_pair(ext_management_system, options)
+    klass.create(key_pair_opts.merge(:resource => ext_management_system))
```

Comment:
NOTE I merged the `:resource` in here in core because it would be the same for every provider implementation

Reply:
thanks for the comment.
It is a little strange that the ems is passed into the function creating the options and yet we have to merge it in here.
But that reduction of duplication does make sense.

Reply:
`ext_management_system` passed in is needed so that we can get an API connection